prog/aws/nic.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require "aws-sdk-ec2"
+class Prog::Aws::Nic < Prog::Base
+  subject_is :nic
+
+  label def create_network_interface
+    network_interface_response = client.create_network_interface({
+      subnet_id: nic.private_subnet.private_subnet_aws_resource.subnet_id,
+      private_ip_address: nic.private_ipv4.network.to_s,
+      ipv_6_prefix_count: 1,
+      groups: [
+        nic.private_subnet.private_subnet_aws_resource.security_group_id
+      ],
+      tag_specifications: tag_specifications("network-interface")
+    })
+    network_interface_id = network_interface_response.network_interface.network_interface_id
+
+    client.assign_ipv_6_addresses({
+      network_interface_id:,
+      ipv_6_address_count: 1
+    })
+
+    nic.update(name: network_interface_id)
+    hop_wait_network_interface_created
+  end
+
+  label def wait_network_interface_created
+    network_interface_response = client.describe_network_interfaces({filters: [{name: "network-interface-id", values: [nic.name]}, {name: "tag:Ubicloud", values: ["true"]}]}).network_interfaces[0]
+    if network_interface_response.status == "available"
+      eip_response = client.allocate_address({
+        domain: "vpc" # Required for VPC-based instances
+      })
```

Comment:
```suggestion
      eip_response = client.allocate_address(domain: "vpc")
```

I think the classic mode is so negligible in this day in age (he says) that we can tighten this up...assuming `domain` is even required.